internal/statemachine/classic_states.go

```diff
@@ -610,6 +617,38 @@ func (stateMachine *StateMachine) addExtraPPAs() (err error) {
 	return nil
 }
 
+// cleanExtraPPAs cleans previously added PPA to the source list
+func (stateMachine *StateMachine) cleanExtraPPAs() (err error) {
+	classicStateMachine := stateMachine.parent.(*ClassicStateMachine)
+
+	sourcesListD := filepath.Join(stateMachine.tempDirs.chroot, "etc", "apt", "sources.list.d")
+
+	for _, ppa := range classicStateMachine.ImageDef.Customization.ExtraPPAs {
+		if helper.PtrBool(ppa.KeepEnabled) {
+			continue
+		}
+		ppaFileName, _ := createPPAInfo(ppa, classicStateMachine.ImageDef.Series)
+
+		ppaFile := filepath.Join(sourcesListD, ppaFileName)
+		err = osRemove(ppaFile)
+		if err != nil {
+			err = fmt.Errorf("Error removing %s: %s", ppaFile, err.Error())
+			return err
+		}
+
+		keyFileName := strings.Replace(ppaFileName, ".list", ".gpg", 1)
+		keyFilePath := filepath.Join(classicStateMachine.tempDirs.chroot,
```

Comment:
Should we also move part of the long filepath listing outside of the loop, like we have for `sourcesListD`? Just purely for consistency and readability.